federation/pkg/federation-controller/util/federated_object.go

```diff
@@ -0,0 +1,63 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"reflect"
+
+	pkg_runtime "k8s.io/kubernetes/pkg/runtime"
+)
+
+const (
+	//TODO: This will be removed once cluster name field is added to ObjectMeta.
+	ClusterNameAnnotation = "federation.io/name"
+)
+
+// TODO: This will be refactored once cluster name field is added to ObjectMeta.
+func GetClusterName(obj pkg_runtime.Object) (string, error) {
+	objectMetaValue := reflect.ValueOf(obj).Elem().FieldByName("ObjectMeta")
+	if !objectMetaValue.IsValid() {
+		return "", fmt.Errorf("no ObjectMeta in the object")
+	}
+	annotationsValue := objectMetaValue.FieldByName("Annotations")
+	if !annotationsValue.IsValid() {
+		return "", fmt.Errorf("no Annotations present")
+	}
+	annotationValue := annotationsValue.MapIndex(reflect.ValueOf(ClusterNameAnnotation))
+	if !annotationValue.IsValid() {
+		return "", fmt.Errorf("annotation is not defined %s", ClusterNameAnnotation)
+	}
+	return annotationValue.String(), nil
+}
+
+// TODO: This will be removed once cluster name field is added to ObjectMeta.
+func SetClusterName(obj pkg_runtime.Object, clusterName string) error {
```

Comment:
Until #28921 is in we need a way to both get information about the origin of the object and set it somehow. SetClusterName is not used in this PR but will be definitely used in other places.